sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1053,7 +1053,7 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
-  val SUBQUERY_REUSE_ENABLED = buildConf("spark.sql.subquery.reuse")
+  val SUBQUERY_REUSE_ENABLED = buildConf("spark.sql.execution.subquery.reuse.enabled")
```

Comment:
now this is inconsistent with the exchange reuse: `spark.sql.exchange.reuse`

cc @rxin @maryannxue

Reply:
It seems we currently have the three patterns for boolean configs;
 - spark.sql.xxx.enabled, e.g., `spark.sql.optimizer.dynamicPartitionPruning.enabled`
 - spark.sql.enableXXX, e.g., `spark.sql.inMemoryColumnarStorage.enableVectorizedReader`
 - the names without `enabled`, e.g., `spark.sql.exchange.reuse`

We need a consistent rule for these boolean configs?